lib/datasource.js

```diff
@@ -468,23 +468,21 @@ DataSource.prototype.defineRelations = function(modelClass, relations) {
   // Set up the relations
   if (relations) {
     Object.keys(relations).forEach(function(rn) {
+      var targetModel;
       var r = relations[rn];
+
       assert(DataSource.relationTypes.indexOf(r.type) !== -1, 'Invalid relation type: ' + r.type);
       assert(isValidRelationName(rn), 'Invalid relation name: ' + rn);
 
-      var targetModel, polymorphicName;
-
       if (r.polymorphic && r.type !== 'belongsTo' && !r.model) {
-        throw new Error(g.f('No model specified for {{polymorphic}} %s: %s', r.type, rn));
+        throw new Error(g.f('Param "model" not specified for {{polymorphic}} %s relation: %s', r.type, rn));
       }
-
-      if (r.polymorphic) {
-        polymorphicName = typeof r.model === 'string' ? r.model : rn;
-        if (typeof r.polymorphic === 'string') {
-          polymorphicName = r.polymorphic;
```

Comment:
this i guess was once added to enable a config like
```js 
relation: {
  someRelation: {
    type: 'hasMany',
    polymorphic: 'someOtherModel'
  }
}
```

but it is not documented and now conflicts with other ways of settings theses relations as well with some other validations in the code